### PR TITLE
[Metrics] Add the OOM failure graph 

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Add the OOM failure graph.

I added under the top section (task/actor) since it is much easier to discover + related to tasks and actors failures. 

In the long term, we can replace this graph to failure type graph (e.g., include application failure, node failure, etc.). 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
LGTM. Maybe also attach a screenshot next time for ui changes?

**Comment by clarng** [comment]
can we get a visual example of the change?

**Comment by rkooo567** [comment]
Screen Shot 2023-04-04 at 1 10 38 AM

See the last image

**Comment by rkooo567** [comment]
Can I get an approval? Also cc @zhe-thoughts this is one of the last milestones to support OOM debugging in Ray 2.4

**Comment by zhe-thoughts** [comment]
@ericl could you approve the PR itself first? 

From my side, I think this is good to cherry pick (we are cutting the release branch soon)

File: dashboard/client/src/pages/metrics/Metrics.tsx [hunk 1]
```diff
@@ -119,6 +119,10 @@ const METRICS_CONFIG: MetricsSectionConfig[] = [
         title: "Active Actors by Name",
         pathParams: "orgId=1&theme=light&panelId=36",
       },
+      {
+        title: "Out of Memory Failure by Name",
```
Comment (ericl) [inline dashboard/client/src/pages/metrics/Metrics.tsx:123]:
```suggestion
        title: "Out of Memory Failures by Name",
```

File: dashboard/modules/metrics/dashboards/default_dashboard_panels.py [hunk 2]
```diff
@@ -239,6 +239,18 @@ def max_plus_pending(max_resource, pending_resource):
             ),
         ],
     ),
+    Panel(
+        id=44,
+        title="Node Out of Memory Failure by Name",
```
Comment (ericl) [inline dashboard/modules/metrics/dashboards/default_dashboard_panels.py:244]:
```suggestion
        title="Node Out of Memory Failures by Name",
```

File: dashboard/modules/metrics/dashboards/default_dashboard_panels.py [hunk 3]
```diff
@@ -239,6 +239,18 @@ def max_plus_pending(max_resource, pending_resource):
...
+        description="The number of tasks and actors killed by Ray Out of Memory killer due to high memory pressure. Metrics are broken down by IP and the name. https://docs.ray.io/en/master/ray-core/scheduling/ray-oom-prevention.html.",
```
Comment (ericl) [inline dashboard/modules/metrics/dashboards/default_dashboard_panels.py:245]:
```suggestion
        description="The number of tasks and actors killed by the Ray Out of Memory killer due to high memory pressure. Metrics are broken down by IP and the name. https://docs.ray.io/en/master/ray-core/scheduling/ray-oom-prevention.html.",
```

File: doc/source/ray-observability/ray-metrics.rst [hunk 4]
```diff
@@ -136,6 +136,9 @@ Ray exports a number of system metrics, which provide introspection into the sta
    * - `ray_placement_groups`
      - `State`
      - Current number of placement groups by state. The State label (e.g., PENDING, CREATED, REMOVED) describes the state of the placement group. See `rpc::PlacementGroupTable <https://github.com/ray-project/ray/blob/e85355b9b593742b4f5cb72cab92051980fa73d3/src/ray/protobuf/gcs.proto#L517>`_ for more information.
+   * - `ray_memory_manager_worker_eviction_total`
+     - `Type`, `Name`
+     - The number of tasks and actors killed by Ray Out of Memory killer (https://docs.ray.io/en/master/ray-core/scheduling/ray-oom-prevention.html) broken down by types (whether it is tasks or actors) and names (name of tasks and actors).
```
Comment (ericl) [inline doc/source/ray-observability/ray-metrics.rst:141]:
```suggestion
     - The number of tasks and actors killed by the Ray Out of Memory killer (https://docs.ray.io/en/master/ray-core/scheduling/ray-oom-prevention.html) broken down by types (whether it is tasks or actors) and names (name of tasks and actors).
```

**Review by ericl — APPROVED** [review]

**Review by zhe-thoughts — APPROVED** [review]
Let's remember to cherry pick this

**Pull request merged**